packages/app-store/_utils/useAddAppMutation.ts

```diff
@@ -47,17 +50,18 @@ function useAddAppMutation(_type: App["type"] | null, options?: Parameters<typeo
     }
 
     const json = await res.json();
+    const externalUrl = /https?:\/\//.test(json.url) && !json.url.startsWith(window.location.origin);
 
     if (!isOmniInstall) {
       gotoUrl(json.url, json.newTab);
-      return;
+      return { setupPending: !externalUrl && !json.url.endsWith("/setup") };
```

Comment:
This is to let the UI know if it should show the successful app installation message.